nsis/src/nsis/NsisLoader.java

```diff
@@ -91,23 +87,23 @@ protected void load(ByteProvider provider, LoadSpec loadSpec, List<Option> optio
 			NsisExecutable ne = NsisExecutable.createNsisExecutable(factory, provider,
 					SectionLayout.FILE);
 			long scriptHeaderOffset = ne.getHeaderOffset();
-			BinaryReader binary_reader = new BinaryReader(provider, /* isLittleEndian= */ true);
-			binary_reader.setPointerIndex(scriptHeaderOffset);
+
 			Address scriptHeaderAddress = program.getAddressFactory().getDefaultAddressSpace()
 					.getAddress(scriptHeaderOffset);
 
-			FileBytes fileBytesHeader = MemoryBlockUtils.createFileBytes(program, provider,
-					scriptHeaderOffset, NsisScriptHeader.getHeaderSize(), monitor);
-			initScriptHeader(fileBytesHeader, scriptHeaderAddress, fileBytesHeader.getSize(),
-					program, ne.getHeaderDataType());
-
-			FileBytes fileBytesBody;
-			byte[] decompressedBytes = ne.getBodyData();
-			ByteArrayProvider uncompressedBytes = new ByteArrayProvider(decompressedBytes);
-			fileBytesBody = MemoryBlockUtils.createFileBytes(program, uncompressedBytes, 0,
-					uncompressedBytes.length(), monitor);
-			initBlockHeaders(program, binary_reader,
-					scriptHeaderAddress.add(NsisScriptHeader.getHeaderSize()), fileBytesBody);
+			InputStream headerInputStream = provider.getInputStream(scriptHeaderOffset);
```

Comment:
When you have to use and close an `InputStream` like this use this pattern:
```
InputStream headerInputStream = null;
try {
  headerInputStream = provider.getInputStream(scriptHeaderOffset);
  initScriptHeader(headerInputStream, scriptHeaderAddress, program,
    ne.getHeaderDataType(), monitor, NsisScriptHeader.getHeaderSize());
}
finally {
  if (headerInputStream != null) {
    headerInputStream.close()
}
```

And at some point Java introduced a cleaner way to to this, and since InputStream implements `AutoClosable` you can use it:

```
try (InputStream headerInputStream = provider.getInputStream(scriptHeaderOffset)) {
  initScriptHeader(headerInputStream, scriptHeaderAddress, program,
    ne.getHeaderDataType(), monitor, NsisScriptHeader.getHeaderSize());
}
```

No need to call close, it will be done automatically.